generate.ml

```diff
@@ -146,11 +146,38 @@ let gen_const typ =
   | _ ->
     Cil.mkCast ~force:false ~e:(Cil.zero ~loc) ~newt:typ
 
+let gen_array_elt (lhost, offset as lval) =
+  match Cil.typeOfLval lval with
+  | TArray (_, e, _, _) ->
+    (* Generate random array index in [0, array_length - 1], just for the last
+       array dimension. *)
+    let offset =
+      match offset with
+      | NoOffset ->
+        Index (Cil.(integer ~loc (Random.int (lenOfArray e))), NoOffset)
+      | _ ->
+        offset
+    in
+    lhost, offset
+  | typ ->
+    let err_msg =
+      Format.asprintf
+        "Expected lvalue of array type, got `%a'."
+        Cil_printer.pp_typ typ
+    in
+    raise (Invalid_argument err_msg)
+
 (* Initialize a local variable to a constant or a parameter. *)
 let gen_local_init lval =
   let gen_param_use typ =
     if not (LvalSet.is_empty !param_lvals) then
       let lval = Utils.random_select_from_set !param_lvals in
+      let lval =
+        if Cil.(isArrayType (typeOfLval lval)) then
+          gen_array_elt lval
+        else
+          lval
+      in
```

Comment:
As this pattern is repeated below, you could factor it out into a function called `gen_use` or something. (I would like `gen_lval_use`, but that's already taken. Feel free to rename things if you find a better name.)

Reply:
Will do. I will guess name the function `gen_lval_elt_use`. Is it OK with you?

Reply:
> Will do. I will guess name the function `gen_lval_elt_use`. Is it OK with you?

Well, my suggestion would not be correct. I guess I will call it `gen_use` then.

Reply:
Oh wait, what about `gen_lval_occurrence`?

Reply:
`gen_lval_occurrence` sounds good, thanks!